tests/fixtures/optimizer/qualify_columns.sql

```diff
@@ -317,6 +317,14 @@ SELECT COALESCE(y.b, z.b) AS b, COALESCE(y.c, z.c) AS c FROM y AS y JOIN z AS z
 SELECT * FROM y JOIN z USING(b, c) WHERE b = 2 AND c = 3;
 SELECT COALESCE(y.b, z.b) AS b, COALESCE(y.c, z.c) AS c FROM y AS y JOIN z AS z ON y.b = z.b AND y.c = z.c WHERE COALESCE(y.b, z.b) = 2 AND COALESCE(y.c, z.c) = 3;
 
+# execute: false
+SELECT b FROM x JOIN t USING(a);
```

Comment:
since it's using, shouldn't the coalesce logic kick in?

Reply:
FWIW if we knew that `b` belongs to both `x` and `t`, the current behaviour is to raise an optimizer error:

```python
>>> from sqlglot.optimizer import optimize
>>> optimize("select x from t1 join t2 using(id)", schema={"t1": {"x": "int", "id": "int"}, "t2": {"x": "int", "id": "int"}}).sql()
# sqlglot.errors.OptimizeError: Unknown table: '' for column 'x'
```

Reply:
Discussed in Slack.